src/hooks/useCart/useCart.tsx

```diff
@@ -235,11 +235,22 @@ export const useCart = (
    */
   const _completeCheckout = useCallback(() => {
     const complete = async (): Promise<void> => {
-      const transactions = Object.values(cart)
-        .filter(({ quantity }) => quantity)
-        .map(({ category, quantity, identifierInputs }) => {
-          return { category, quantity, identifierInputs };
-        });
+      const allIdentifierInputs: IdentifierInput[] = [];
+      let transactions: any = [];
+      try {
+        transactions = Object.values(cart)
+          .filter(({ quantity }) => quantity)
+          .map(({ category, quantity, identifierInputs }) => {
+            allIdentifierInputs.push(...identifierInputs);
+            validateIdentifierInputs(allIdentifierInputs);
+            return { category, quantity, identifierInputs };
+          });
```

Comment:
Notice you are still returning the uncleaned identifierInputs here (i.e. `return { category, quantity, identifierInputs };`). In addition, this logic will result to the validation of identiferInputs of the previous transactions repeatedly. 

how about you try smth like this:

```suggestion
      try {
        transactions = Object.values(cart)
          .filter(({ quantity }) => quantity)
          .map(({ category, quantity, identifierInputs }) => {
            const cleanedIdentifierInputs = validateIdentifierInputs(identifierInputs);
            return { category, quantity, identifierInputs: cleanedIdentifierInputs };
          });
```
do add a test in `useCart.test.tsx` to cover this change, and make sure whitespaced inputs are not passed in